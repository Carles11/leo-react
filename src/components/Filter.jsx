import React from 'react'
import PropTypes from 'prop-types'

class Filter extends React.PureComponent {
	state = {
		visible: false,
	}

	static propTypes = {
		filter: PropTypes.object.isRequired,
		handleFilter: PropTypes.func.isRequired,
	}

	handleVisibility = () => {
		this.setState(prevState => ({ visible: !prevState.visible }))
	}

	handleFilter = e => {
		this.setState({ visible: false }, this.props.handleFilter(e))
	}

	render() {
		const { filter } = this.props
		const { visible } = this.state
		const active = visible ? 'active' : ''

		return (
			<div className="app-filter">
				<div className="app-filter-label">
					<p>Galeria de Images de la edición del </p>
					<button id="btn_visibility" onClick={this.handleVisibility}>
						<span>{'Año'.toUpperCase()}</span>
						<span className="icon-down" />
					</button>
					<ul className={`app-filter-list ${active}`}>
						{Object.keys(filter).map(k => (
							<li className="app-filter-item" key={k}>
								<button
									id={`btn_${k}`}
									data-year={k}
									data-count={filter[k]}
									onClick={this.handleFilter}
									disabled={!visible}
								>
									{k}
								</button>
							</li>
						))}
						<li className="app-filter-item">
							<button onClick={this.handleFilter} disabled={!visible}>
								TODAS
							</button>
						</li>
					</ul>
				</div>
			</div>
		)
	}
}

export default Filter
