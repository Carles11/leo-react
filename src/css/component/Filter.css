.app-filter {
  position: absolute;
  top: 6rem;
  left: -277px;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  opacity: 0.99;
  transition-property: left; 
  transition-duration: 0.3s;
}

.app-filter:hover {
  left: 0;
}

.app-filter:hover .app-filter-label span[class^=icon-] {
  transform: rotate(360deg)
}

.app-filter-label {
  position: relative;
  display: flex;
  background: var(--menuBg);
  color: var(--textColor);
  font-family: Bold;
  font-size: var(--textSize);
  text-decoration: none;
  box-sizing: border-box;
  transition-property: background;
  transition-duration: 0.5s;
  z-index: 0;
  border: 1px solid #eeeeee;
}

.app-filter-label > p {
  font-family: Lighter;
  padding: 1rem 0 1rem 1rem;
  margin: 0;
}

.app-filter-label > button {
  border: none;
  font-family: Bold;
  font-size: var(--textSize);
  background: none;
  outline: none;
  cursor: pointer;
  display: flex;
  align-items: center;
}

.app-filter-label span[class^=icon-] {
  display: block;
  transform: rotate(270deg);
  transition-property: transform;
  transition-duration: 0.5s; 
  margin-left: 0.5rem;
}

.app-filter-list {
  visibility: hidden;
  position: absolute;
  top: 2rem;
  list-style: none;
  width: 100%;
  padding: 0;
  margin: 0;
  transition-property: top;
  transition-duration: 0.5s;
}

.app-filter-list.active {
  visibility: visible;
  top: -1px;
}

.app-filter-item {
  position: relative;
  width: 100%;
  padding: 0;
  box-sizing: border-box;
}

.app-filter-item > button {
  display: block;
  width: 100%;
  background: var(--menuBg);
  font-size: var(--textSize);
  font-family: Lighter;
  text-align: left;
  border: none;
  padding: 1.3rem;
  border: 1px solid #eeeeee;
  border-bottom: none;
  cursor: pointer;
  transition-property: background;
  transition-duration: 0.5s; 
}

.app-filter-item > button:hover {
  background: white;
}
.app-filter-item:last-of-type > button {
  border: 1px solid #eeeeee;
}

.app-filter-item > button[data-count]:not([data-count=""]):after {
  margin-left: 0.5rem;
  content: "(" attr(data-count) ")";
  color: #cccccc;
}
