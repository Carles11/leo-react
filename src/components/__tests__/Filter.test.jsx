import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import Filter from '../Filter'

const properties = {
  filter: {
    2018: 1,
  },
  handleFilter: jest.fn(),
}

function setup() {
  const props = properties
  const enzymeWrapper = mount(
    <MemoryRouter initialEntries={['/test']}>
      <Filter {...props} />
    </MemoryRouter>,
  )

  return {
    props,
    enzymeWrapper,
  }
}

describe('<Filter />', () => {
  it('should render itself', () => {
    const component = shallow(<Filter {...properties} />)
    expect(component).toMatchSnapshot()
  })

  it('should call the handleFilter prop method', () => {
    const { enzymeWrapper, props } = setup()
    const mockedEvent = { test: true }
    enzymeWrapper.find('#btn_visibility').simulate('click')
    expect(props.handleFilter.mock.calls.length).toEqual(0)
    enzymeWrapper.find('#btn_2018').simulate('click')
    expect(props.handleFilter.mock.calls.length).toEqual(1)
  })

  it('should render all the filter items', () => {
    properties.filter[2019] = 2
    const { enzymeWrapper, props } = setup()
    expect(enzymeWrapper.find('.app-filter-item button').length).toEqual(3)
  })
})
