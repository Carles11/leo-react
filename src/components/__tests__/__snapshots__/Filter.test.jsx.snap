// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render itself 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    filter={
        Object {
            "2018": 1,
          }
    }
    handleFilter={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="app-filter-label"
>
        <p>
                Galeria de Images de la edición del 
        </p>
        <button
                id="btn_visibility"
                onClick={[Function]}
        >
                <span>
                        AÑO
                </span>
                <span
                        className="icon-down"
                />
        </button>
        <ul
                className="app-filter-list "
        >
                <li
                        className="app-filter-item"
                >
                        <button
                                data-count={1}
                                data-year="2018"
                                disabled={true}
                                id="btn_2018"
                                onClick={[Function]}
                        >
                                2018
                        </button>
                </li>
                <li
                        className="app-filter-item"
                >
                        <button
                                disabled={true}
                                onClick={[Function]}
                        >
                                TODAS
                        </button>
                </li>
        </ul>
</div>,
      "className": "app-filter",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Galeria de Images de la edición del 
</p>,
          <button
            id="btn_visibility"
            onClick={[Function]}
>
            <span>
                        AÑO
            </span>
            <span
                        className="icon-down"
            />
</button>,
          <ul
            className="app-filter-list "
>
            <li
                        className="app-filter-item"
            >
                        <button
                                    data-count={1}
                                    data-year="2018"
                                    disabled={true}
                                    id="btn_2018"
                                    onClick={[Function]}
                        >
                                    2018
                        </button>
            </li>
            <li
                        className="app-filter-item"
            >
                        <button
                                    disabled={true}
                                    onClick={[Function]}
                        >
                                    TODAS
                        </button>
            </li>
</ul>,
        ],
        "className": "app-filter-label",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Galeria de Images de la edición del ",
          },
          "ref": null,
          "rendered": "Galeria de Images de la edición del ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                AÑO
</span>,
              <span
                className="icon-down"
/>,
            ],
            "id": "btn_visibility",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AÑO",
              },
              "ref": null,
              "rendered": "AÑO",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon-down",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <li
                  className="app-filter-item"
>
                  <button
                                    data-count={1}
                                    data-year="2018"
                                    disabled={true}
                                    id="btn_2018"
                                    onClick={[Function]}
                  >
                                    2018
                  </button>
</li>,
              ],
              <li
                className="app-filter-item"
>
                <button
                                disabled={true}
                                onClick={[Function]}
                >
                                TODAS
                </button>
</li>,
            ],
            "className": "app-filter-list ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2018",
              "nodeType": "host",
              "props": Object {
                "children": <button
                  data-count={1}
                  data-year="2018"
                  disabled={true}
                  id="btn_2018"
                  onClick={[Function]}
>
                  2018
</button>,
                "className": "app-filter-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2018",
                  "data-count": 1,
                  "data-year": "2018",
                  "disabled": true,
                  "id": "btn_2018",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "2018",
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  disabled={true}
                  onClick={[Function]}
>
                  TODAS
</button>,
                "className": "app-filter-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TODAS",
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "TODAS",
                "type": "button",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="app-filter-label"
>
          <p>
                    Galeria de Images de la edición del 
          </p>
          <button
                    id="btn_visibility"
                    onClick={[Function]}
          >
                    <span>
                              AÑO
                    </span>
                    <span
                              className="icon-down"
                    />
          </button>
          <ul
                    className="app-filter-list "
          >
                    <li
                              className="app-filter-item"
                    >
                              <button
                                        data-count={1}
                                        data-year="2018"
                                        disabled={true}
                                        id="btn_2018"
                                        onClick={[Function]}
                              >
                                        2018
                              </button>
                    </li>
                    <li
                              className="app-filter-item"
                    >
                              <button
                                        disabled={true}
                                        onClick={[Function]}
                              >
                                        TODAS
                              </button>
                    </li>
          </ul>
</div>,
        "className": "app-filter",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Galeria de Images de la edición del 
</p>,
            <button
              id="btn_visibility"
              onClick={[Function]}
>
              <span>
                            AÑO
              </span>
              <span
                            className="icon-down"
              />
</button>,
            <ul
              className="app-filter-list "
>
              <li
                            className="app-filter-item"
              >
                            <button
                                          data-count={1}
                                          data-year="2018"
                                          disabled={true}
                                          id="btn_2018"
                                          onClick={[Function]}
                            >
                                          2018
                            </button>
              </li>
              <li
                            className="app-filter-item"
              >
                            <button
                                          disabled={true}
                                          onClick={[Function]}
                            >
                                          TODAS
                            </button>
              </li>
</ul>,
          ],
          "className": "app-filter-label",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Galeria de Images de la edición del ",
            },
            "ref": null,
            "rendered": "Galeria de Images de la edición del ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  AÑO
</span>,
                <span
                  className="icon-down"
/>,
              ],
              "id": "btn_visibility",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AÑO",
                },
                "ref": null,
                "rendered": "AÑO",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-down",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  <li
                    className="app-filter-item"
>
                    <button
                                        data-count={1}
                                        data-year="2018"
                                        disabled={true}
                                        id="btn_2018"
                                        onClick={[Function]}
                    >
                                        2018
                    </button>
</li>,
                ],
                <li
                  className="app-filter-item"
>
                  <button
                                    disabled={true}
                                    onClick={[Function]}
                  >
                                    TODAS
                  </button>
</li>,
              ],
              "className": "app-filter-list ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "2018",
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    data-count={1}
                    data-year="2018"
                    disabled={true}
                    id="btn_2018"
                    onClick={[Function]}
>
                    2018
</button>,
                  "className": "app-filter-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2018",
                    "data-count": 1,
                    "data-year": "2018",
                    "disabled": true,
                    "id": "btn_2018",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "2018",
                  "type": "button",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    disabled={true}
                    onClick={[Function]}
>
                    TODAS
</button>,
                  "className": "app-filter-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TODAS",
                    "disabled": true,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "TODAS",
                  "type": "button",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
